### DISCUSSION TITLE
Seg Fault after creating and trying to visualize a mesh

### DISCUSSION THREAD

**Discussion by SofianeHamlaoui** [pyvista/pyvista#4555] [Q&A]
So, 
I'm getting the necessary data to make a mesh for Pyvista from Threejs BufferGeometry and using it in Pyvista. 

I'm having a function that converts the threejs coordinates to Pyvista : 

* vertex_positions 

* vertex_normals

* uv_coordinates

* triangle_indices 

And this is the code i'm running to show and visualize the mesh : 

```
pyvista_mesh = convert_threejs_to_pyvista(parsed_data)

p = pv.Plotter()
p.show_axes()
p.add_mesh(pyvista_mesh)
p.show()
```

And I'm having a Seg Fault while trying this : 

My output : 

```bash
#-  printing pyvista_mesh -#
PolyData (0x1d6af39e260)
  N Cells:    6
  N Points:   24
  N Strips:   0
  X Bounds:   1.845e+06, 1.845e+06
  Y Bounds:   5.175e+06, 5.175e+06
  Z Bounds:   1.442e+02, 1.942e+02
  N Arrays:   0
Segmentation fault
```
Anyone faced this issue or already tried doing this ? 

Thanks you in advance.

**Comment by banesullivan** [comment]
Could you share the source data so that we can reproduce this?

**Comment by SofianeHamlaoui** [comment]
> Could you share the source data so that we can reproduce this?

Of course, so the data I obtained from threejs BufferGeometry is [Here](https://pastebin.com/raw/kJdE8kfF) ( 3Data.json ) 

The full code used  : 

```python
import pyvista as pv
import numpy as np
import json
from pythreejs import Geometry

f = open("3Data.json")

parsed_data = json.load(f)

def convert_threejs_to_pyvista(geometry):
    # Extract vertex positions
    positions = np.array(geometry["vertices"])

    # Extract face indices
    indices = np.hstack(geometry["triangles"])

    # Extract vertex normals (optional)
    normals = None
    if "normal" in geometry["normals"]:
        normals = np.array(geometry["normals"])

    # Extract UV coordinates (optional)
    uvs = None
    if "uv" in geometry["uvs"]:
        uvs = np.array(geometry["uvs"])

    # Create PyVista mesh
    mesh = pv.PolyData(positions, indices)

    if normals is not None:
        mesh["normals"] = normals

    if uvs is not None:
        mesh["texture_coords"] = uvs

    return mesh

pyvista_mesh = convert_threejs_to_pyvista(parsed_data)

print(pyvista_mesh)
p = pv.Plotter()
p.add_mesh(pyvista_mesh)
p.show()

```

**Comment by edabor** [comment]
@SofianeHamlaoui Here are the changes for your code to work:
![image](https://github.com/pyvista/pyvista/assets/117822458/5bc0d31d-7cbd-4671-9c1d-bfd23f03292b)

The segfault was caused by the connectivity which was not filled correctly.
I also changed a few things for **point_data** to be correctly set in your **polydata**.

However, I am afraid the plot won't show you anything relevant, since the triangles are very elongated in your data.

Full code:
```python
import pyvista as pv
import numpy as np
import json
from pythreejs import Geometry

f = open("3Data.json")

parsed_data = json.load(f)


def convert_threejs_to_pyvista(geometry):
    # Extract vertex positions
    positions = np.array(geometry["vertices"])

    # Extract face indices
    indices = np.hstack(geometry["triangles"])
    cells = {pv.CellType.TRIANGLE: indices}

    # Extract vertex normals (optional)
    normals = None
    if "normals" in geometry:
        normals = np.array(geometry["normals"])

    # Extract UV coordinates (optional)
    uvs = None
    if "uvs" in geometry:
        uvs = np.array(geometry["uvs"])

    # Create PyVista mesh
    mesh = pv.UnstructuredGrid(cells, positions).extract_surface()

    if normals is not None:
        mesh.point_data["normals"] = normals

    if uvs is not None:
        mesh.point_data["texture_coords"] = uvs

    return mesh


pyvista_mesh = convert_threejs_to_pyvista(parsed_data)

print(pyvista_mesh)
p = pv.Plotter()
p.add_mesh(pyvista_mesh)
p.show()

```

**Reply by SofianeHamlaoui** [reply]
Okey, thank you for this help and solution. 

For the data that's the only way I was able to get them, is there a way to compresse or shorten them ? 

Thank you in advance.

**Reply by edabor** [reply]
You can directly modify you polydata points coordinates:
https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.DataSet.points.html#pyvista.DataSet.points

Or you can use a scaling operation:
https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.DataSet.scale.html#pyvista.DataSet.scale

**Reply by SofianeHamlaoui** [reply]
thank you for those suggestions, I think the issue here is the data I get from Threejs ! because when I do the same thing with the extracted vertices and faces from a Pyvista Example Mesh it works perfectly !
